compiler/rustc_mir/src/transform/check_unsafety.rs

```diff
@@ -132,6 +132,24 @@ impl<'a, 'tcx> Visitor<'tcx> for UnsafetyChecker<'a, 'tcx> {
                     self.register_violations(&violations, &unsafe_blocks);
                 }
             },
+            Rvalue::Cast(kind, source, mir_cast_ty) => {
+                if let CastKind::Pointer(ty::adjustment::PointerCast::Unsize) = kind {
+                    let mir_source_ty = match *source {
```

Comment:
You should be able to use `Operand::ty` here, right?

Reply:
I'm not totally sure, since this is earlier than codegen phase, if i'm not misunderstanding it. I'll take a lot.

Reply:
This has nothing to do with codegen. I think you basically inlined `Operand::ty` here.